spectator-ext-aws/src/main/java/com/netflix/spectator/aws/SpectatorRequestMetricCollector.java

```diff
@@ -156,6 +203,10 @@ private Id metricId(String metric, Map<String, String> tags) {
       allTags.put(tag.getName(), tag.getValue(metrics).orElse(UNKNOWN));
     }
     allTags.put(TAG_REQUEST_TYPE, request.getOriginalRequest().getClass().getSimpleName());
+    String contextTagValue = request.getHandlerContext(handlerContextKey);
+    if (contextTagValue != null) {
```

Comment:
Should the gauges be contingent on the context being set? Otherwise existing users might start getting the gauges for pool metrics and not realize that the values reported do not make sense when they have multiple pools in use.

Reply:
Yeah, that makes sense: https://github.com/Netflix/spectator/pull/918/commits/5b77464959b3aa7166c30f601b1eead476cc8750